webscrapping101/scrape.markdown

```diff
@@ -0,0 +1,313 @@
+# Web Scraping 101
+
+<img src="image/header.jpg" width="500" height="300">
+
+## Background
+
+We saw the importance of data in the modern world in the last decade, from making business decisions to building groundbreacking AI system, massive datasets are used in to provide valuable insights in all fields around the world providing. However, gathering data are often times a challenging task, as it is time consuming and expensive.
+
+A place full of free data is the web, thus utilizing the web for gathering data has become a common practice. We use a powerful technique called Web scraping for such task, automating the manual extraction of data and information from websites online. Web scraping often comes in a two step process, collect the data in raw forms then parse the data for further use.
+
+This guide will we will walkthrough an example of Python's HTTPX for client request and BeautifulSoup to parse HTML of a basic, static, non-blocking webpage.
+
+A [Jupyter Notebook](./scrape.ipynb) file is also provided for running the codes presented in this Markdown article for you to experiment the example we are using here.
+
+1. HTTP Requests and web fundamentals.
+2. HTML and javascripts
+3. Python libraries
+4. Web Scraping - Umich EECS
+5. Clean and store information.
+6. Dealing with popups
+7. Other things to consider.
+
+## Web requests
+
+Before diving into web scraping, it is important to understand HTTP fundamentals. The modern websites
+are served through the HTTP protocol, which is used for transmitting hypertext requests and information
+between servers and browsers. We, as clients, send requests to the webite (servers) for resources. The server
+processes our request and reply with a response of cooresponding web data (or error message in the event of a failure )
+
+![exchange](./image/exchange.png)
+
+In the process of web scraping, we primarily deal with HTTP requests and responses. These are the fundamental components of data communication on the web. Let's take a look into their structure and relevance to web scraping.
+
+### HTTP Requests
+
+An HTTP request is composed of three main parts:
+
+1. \*Method: one of several types that define the kind of action to be performed. eg.
+    - GET: requests a resource.
+    - POST: requests a resource by sending additional data.
+    - HEAD: requests meta information about a resource, such as its last update time.
+2. Headers: provide metadata about our request.
+3. Location: specifies the resource we aim to retrieve. They are defined by URL (Uniform Resource Locator)
+
+![url](./image/urlstructure.png)
```

Comment:
I think it's worth spending a little time here explaining the search parameters, how they are formatted, and how the subfolder/slug usually relates to the file tree in the server